docs/source/conf.py

```diff
@@ -68,44 +68,125 @@
 # General configuration
 # ---------------------
 
+# Add any paths that contain templates here, relative to this directory.
+templates_path = config["sphinx"]["templates_path"]
+# The master toctree document.
+master_doc = config["sphinx"]["master_doc"]
+# General substitutions.
+project = config["sphinx"]["project"]
+copyright = config["sphinx"]["copyright"]
+# ghissue config
+github_project_url = config["sphinx"]["github_project_url"]
+# The suffix of source filenames.
+source_suffix = config["sphinx"]["source_suffix"]
+# Exclude these glob-style patterns when looking for source files. They are
+# relative to the source/ directory.
+exclude_patterns = config["sphinx"]["exclude_patterns"]
+# The name of the Pygments (syntax highlighting) style to use.
+pygments_style = config["sphinx"]["pygments_style"]
 # Add any Sphinx extension module names here, as strings. They can be extensions
 # coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
-extensions = [
-    "sphinx.ext.autodoc",
-    "sphinx.ext.autosummary",
-    "sphinx.ext.doctest",
-    "sphinx.ext.inheritance_diagram",
-    "sphinx.ext.intersphinx",
-    "sphinx.ext.graphviz",
-    "sphinxcontrib.jquery",
-    "IPython.sphinxext.ipython_console_highlighting",
-    "IPython.sphinxext.ipython_directive",
-    "sphinx.ext.napoleon",  # to preprocess docstrings
-    "github",  # for easy GitHub links
-    "magics",
-    "configtraits",
+extensions = config["sphinx"]["extensions"]
+# Set the default role so we can use `foo` instead of ``foo``
+default_role = config["sphinx"]["default_role"]
+modindex_common_prefix = config["sphinx"]["modindex_common_prefix"]
+
+intersphinx_mapping = config["intersphinx_mapping"]
+for k, v in intersphinx_mapping.items():
+    intersphinx_mapping[k] = tuple(
+        [intersphinx_mapping[k]["url"], intersphinx_mapping[k]["fallback"]]
+    )
```

Comment:
Obviously here we need some workaround, and updating locals won't work.

Reply:
thought on publishing (and maintaining an up to date) intersphinx_mapping python package that just contain standard urls for most of the scientific Python stack ? This has no reasons to differ between proejcts right ?

Reply:
I guess that works only if all the packages always point to latest/stable, but that woudl be a good default I think.